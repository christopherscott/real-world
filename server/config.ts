const {
  NODE_ENV,
  PRISMA_ENDPOINT,
  PRISMA_TEST_ENDPOINT,
  PRISMA_SECRET,
  APP_SECRET,
  TMDB_KEY,
  OMDB_KEY,
} = process.env;

const isTest = NODE_ENV === 'test';

export default {
  PRISMA_ENDPOINT: isTest ? 'http://prisma.com' : PRISMA_ENDPOINT,
  PRISMA_TEST_ENDPOINT: isTest ? 'http://prisma.com' : PRISMA_TEST_ENDPOINT,
  PRISMA_SECRET: isTest ? '123' : PRISMA_SECRET,
  APP_SECRET: isTest ? '123' : APP_SECRET,
  TMDB_KEY: isTest ? '123' : TMDB_KEY,
  OMDB_KEY: isTest ? '123' : OMDB_KEY,
};
