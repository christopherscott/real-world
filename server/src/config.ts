export const PRISMA_ENDPOINT =
  process.env.NODE_ENV === 'test'
    ? 'http://localhost:4467'
    : process.env.PRISMA_ENDPOINT;

export const PRISMA_SECRET = process.env.PRISMA_SECRET;
