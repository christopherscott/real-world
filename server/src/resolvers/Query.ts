import { Context } from '../utils';

const posts = (parent, args, context: Context, info) => {
  return context.db.query.posts({}, info);
};

const post = (parent, { id }: { id: string }, context: Context, info) => {
  return context.db.query.post({ where: { id } }, info);
};

const search = (
  parent,
  { search }: { search?: string },
  context: Context,
  info
) => {
  return context.db.query.posts(
    {
      where: { OR: [{ text_contains: search }, { title_contains: search }] },
    },
    info
  );
};

export default {
  posts,
  post,
  search,
};
