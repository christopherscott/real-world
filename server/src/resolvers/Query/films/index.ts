import { Context } from '../../../utils';
import { Film } from '../../../generated/types';

export const films = (parent, args, context: Context, info) => {
  // Order by multiple fields is not supported yet
  // https://github.com/prismagraphql/prisma/issues/62
  return context.db.query.films({ orderBy: 'rating_DESC' }, info);
};

export const film = (
  parent,
  { id }: { id: Film['id'] },
  context: Context,
  info
) => {
  return context.db.query.film({ where: { id } }, info);
};
