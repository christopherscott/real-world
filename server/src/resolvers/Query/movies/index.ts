import { Context } from '../../../utils';
import { ID_Input } from '../../../generated/prisma';

export const movies = (parent, args, context: Context, info) => {
  // Order by multiple fields is not supported yet
  // https://github.com/prismagraphql/prisma/issues/62
  return context.db.query.movies({ orderBy: 'popularity_DESC' }, info);
};

export const movie = (
  parent,
  { id }: { id: ID_Input },
  context: Context,
  info
) => {
  return context.db.query.movie({ where: { id } }, info);
};
