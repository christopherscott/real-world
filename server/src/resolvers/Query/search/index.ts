import { Context } from '../../../utils';

// TODO: Use elasticsearch
export const search = async (
  parent,
  { search }: { search: string },
  context: Context,
  info
) => {
  const movies = await context.db.query.movies(
    {
      where: {
        OR: [{ title_contains: search }, { overview_contains: search }],
      },
    },
    info
  );

  // const comments = await context.db.query.comments(
  //   {
  //     where: { OR: [{ text_contains: search }] },
  //   },
  //   info
  // );

  // return [...films, ...comments];

  return movies;
};
