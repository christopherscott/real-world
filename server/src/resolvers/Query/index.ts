import * as search from './search';
import * as movies from './movies';

export default {
  ...search,
  ...movies,
};
