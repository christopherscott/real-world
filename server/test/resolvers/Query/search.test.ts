import { map, omit, length } from 'ramda';
import { context, db, clean } from '../../config';
import fixtures from '../../fixtures';
import { search } from '../../../src/resolvers/Query/search';

afterEach(async () => {
  await clean();
});

test('should return movies matched by search word', async () => {
  const films = await Promise.all(
    map(data => db.mutation.createMovie({ data }, '{ id }'), fixtures.movies)
  );

  const result = await search(null, { search: 'IMF' }, context, null);

  expect(length(result)).toBe(1);

  expect(map(omit(['id']), result)).toMatchSnapshot();
});
