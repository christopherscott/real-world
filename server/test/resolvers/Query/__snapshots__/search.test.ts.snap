// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return movies matched by search word 1`] = `
Array [
  Object {
    "actors": "Tom Cruise, Rebecca Ferguson, Henry Cavill, Michelle Monaghan",
    "director": "Christopher McQuarrie",
    "overview": "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
    "popularity": 130.91,
    "poster": "posters/80PWnSTkygi3QWWmJ3hrAwqvLnO.jpg",
    "releaseDate": "2018-07-25T00:00:00.000Z",
    "runtime": "1970-01-01T02:27:00.000Z",
    "title": "Mission: Impossible - Fallout",
  },
]
`;
